src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
+        {
+            // If it is a document selection, we'll retrieve the text from the document
+            ExecuteDocumentSelectionParams docRequest = request as ExecuteDocumentSelectionParams;
+            if (docRequest != null)
+            {
+                // Get the document from the parameters
+                ExecuteDocumentSelectionParams documentRequest = (ExecuteDocumentSelectionParams) request;
```

Comment:
You should use `docRequest` since you already cast it and validated it's not null - this 2nd declaration is redundant